diff --git a/Makefile b/Makefile
index 8b70ef7..e71b403 100644
--- a/Makefile
+++ b/Makefile
@@ -6,16 +6,11 @@ include $(THEOS)/makefiles/common.mk
 
 TWEAK_NAME = DouX
 
-DouX_LIBRARIES = JGProgressHUD
-DouX_FILES = Tweak.x $(wildcard hooks/*.x) DouXDownload.m DouXManager.m DouXMultipleDownload.m SecurityViewController.m VaultMediaItem.m VaultManager.m VaultViewController.m PhotoViewController.m CreatorFilterViewController.m ContentTypeFilterViewController.m FilterViewController.m $(wildcard Settings/*.m)
+DouX_FILES = Tweak.x $(wildcard hooks/*.x) DouXDownload.m DouXManager.m DouXMultipleDownload.m SecurityViewController.m VaultMediaItem.m VaultManager.m VaultViewController.m PhotoViewController.m CreatorFilterViewController.m ContentTypeFilterViewController.m FilterViewController.m $(wildcard Settings/*.m) $(wildcard libs/JGProgressHUD/*.m)
 DouX_FRAMEWORKS = UIKit Foundation CoreGraphics Photos CoreServices SystemConfiguration SafariServices Security QuartzCore Photos
 DouX_CFLAGS = -fobjc-arc -Wno-unused-variable -Wno-unused-value -Wno-deprecated-declarations -Wno-nullability-completeness -Wno-unused-function -Wno-incompatible-pointer-types -I./libs/JGProgressHUD
 
 include $(THEOS_MAKE_PATH)/tweak.mk
 
 # --- Add this line at the end ---
-include $(THEOS_MAKE_PATH)/aggregate.mk
-
-before-all::
-	@$(MAKE) -C libs/JGProgressHUD THEOS_PACKAGE_SCHEME=$(THEOS_PACKAGE_SCHEME)
-	@$(MAKE) -C libs/JGProgressHUD stage THEOS_PACKAGE_SCHEME=$(THEOS_PACKAGE_SCHEME)
\ No newline at end of file
+include $(THEOS_MAKE_PATH)/aggregate.mk
\ No newline at end of file
diff --git a/control b/control
index 707b408..7daeafd 100644
--- a/control
+++ b/control
@@ -6,5 +6,5 @@ Description: A collection of enhancements for TikTok.
 Maintainer: kunihir0
 Author: kunihir0
 Section: Tweaks
-Depends: mobilesubstrate,com.kunihir0.jgprogresshud (>= 0.0.3)
+Depends: mobilesubstrate
 Conflicts: com.kunihir0.bhtiktok++,com.raulsaeed.bhtiktok++,com.bandarhl.bhtiktok
